src/components/genes/Genes.js

```diff
@@ -1,49 +1,34 @@
-import React from 'react';
+import React, { useEffect, useCallback } from 'react';
+import { RadioList } from '../selectable-table/index';
 
-export default class Genes extends React.Component {
-  constructor(props) {
-    super(props);
+export default function Genes(props) {
+  const {
+    setSelectedGene,
+    genesSelected,
+    clearPleaseWait,
+  } = props;
 
-    this.Radio = this.Radio.bind(this);
-    this.handleInputChange = this.handleInputChange.bind(this);
-  }
-
-  handleInputChange(event) {
-    const { target } = event;
-    const { name } = target;
-    const { setSelectedGene } = this.props;
-    setSelectedGene(name);
-  }
-
-  Radio(name, value) {
-    return (
-      <div key={name}>
-        <input
-          type="radio"
-          className="radio"
-          name={name}
-          onChange={this.handleInputChange}
-          checked={value}
-        />
-        &nbsp;{name}
-      </div>
-    );
-  }
-
-  render() {
-    const { genesSelected, clearPleaseWait } = this.props;
+  useEffect(() => {
     if (clearPleaseWait && genesSelected) {
       clearPleaseWait('genes');
     }
-    const radioButtons = Object.entries(genesSelected).sort(
-      (a, b) => a[0].localeCompare(b[0]),
-    ).map(
-      ([geneId, value]) => this.Radio(geneId, value),
-    );
-    return (
-      <>
-        {radioButtons}
-      </>
-    );
-  }
+  }, [clearPleaseWait, genesSelected]);
+
+  const onChange = useCallback((selection) => {
+    if (selection && selection.name) {
+      setSelectedGene(selection.name);
+    }
+  }, [setSelectedGene]);
+
+  const data = Object.entries(genesSelected).sort(
+    (a, b) => a[0].localeCompare(b[0]),
+  ).map(
+    ([name, value]) => ({ name, value }),
```

Comment:
(I see that the sort and localeCompare are carried over, but we don't have those on factors... Unless the motivation for the difference is clear to you, can file an issue asking that it be unified?)

Reply:
I am not sure, but maybe the difference is that factors should remain in its initial ordering because there are fewer entries ("cluster" and "subcluster") and they may correspond to levels of a hierarchy? Whereas for genes, there may be hundreds, and an alphabetical ordering makes more sense?